### PR TITLE
Refactored json encoding/decoding

### PR DESCRIPTION
**Pull request by GRYE**

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -12,7 +12,7 @@ mvpSettings {
 otherNodes = [
   {
     host = "0.0.0.0",
-    port = 7777
+    port = 7778
```
Comment (Bromel777) [inline src/main/resources/application.conf]: Please, don't change port)

Comment (GRYE) [reply src/main/resources/application.conf]: Fixed

File: src/main/scala/mvp/local/messageHolder/UserMessage.scala [hunk 2]
```diff
@@ -17,23 +16,20 @@ case class UserMessage(message: String, metadata: Array[Byte], sender: Array[Byt
...
-    msgNum <- c.downField("msgNum").as[Int]
-  } yield UserMessage(
-    message,
-    Base16.decode(metadata).getOrElse(Array.emptyByteArray),
-    Base16.decode(sender).getOrElse(Array.emptyByteArray),
-    None,
-    msgNum
-  )
+  implicit val decodeUserMessage: Decoder[UserMessage] =
+    Decoder.forProduct4[String, String, String, Int, UserMessage]("message", "metadata", "sender", "msgNub") {
```
Comment (Bromel777) [inline src/main/scala/mvp/local/messageHolder/UserMessage.scala]: msgNum, not msgNub

Comment (GRYE) [reply src/main/scala/mvp/local/messageHolder/UserMessage.scala]: Fixed

**Pull request merged**